src/hooks/useWindowWidth.ts

```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+
+export const useWindowWidth = () => {
+  const [windowWidth, setWindowWidth] = React.useState<number>(0)
+
+  function handleResize() {
+    setWindowWidth(window.outerWidth)
+  }
+  React.useEffect(() => {
+    window.addEventListener('resize', handleResize)
+
+    handleResize()
+
+    return () => window.removeEventListener('resize', handleResize)
+  }, [])
```

Comment:
While this might be necessay, I suggest you to crete a useLessThanBreakPoint hook.

that utilizes theme.breakpoints.down(brekpoint) to get isLessThanSm. This would prevent too many re renders.